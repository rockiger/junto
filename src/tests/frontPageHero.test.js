import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import FrontPageHero from '../components/frontPageHero';

describe('FrontPageHero', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<FrontPageHero />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('has a valid snapshot', () => {
        const component = renderer.create(<FrontPageHero />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});