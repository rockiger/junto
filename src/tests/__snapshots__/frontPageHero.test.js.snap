// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPageHero has a valid snapshot 1`] = `
<div
  className="hero"
>
  <div
    className="hero-logo"
  >
    <img
      alt="Awiki Logo"
      src="logo.svg"
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </div>
  <h1
    className="hero-title"
  >
    Welcome to Awiki!
  </h1>
  <p
    className="hero-description"
  />
  <div
    className="hero-row"
  >
    <button
      className="hero-heroButon"
    >
      <svg
        height="24px"
        version="1.1"
        viewBox="0 0 48 48"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M24 9.5c3.54 0 6.71 1.22 9.21 3.6l6.85-6.85C35.9 2.38 30.47 0 24 0 14.62 0 6.51 5.38 2.56 13.22l7.98 6.19C12.43 13.72 17.74 9.5 24 9.5z"
            fill="#EA4335"
          />
          <path
            d="M46.98 24.55c0-1.57-.15-3.09-.38-4.55H24v9.02h12.94c-.58 2.96-2.26 5.48-4.78 7.18l7.73 6c4.51-4.18 7.09-10.36 7.09-17.65z"
            fill="#4285F4"
          />
          <path
            d="M10.53 28.59c-.48-1.45-.76-2.99-.76-4.59s.27-3.14.76-4.59l-7.98-6.19C.92 16.46 0 20.12 0 24c0 3.88.92 7.54 2.56 10.78l7.97-6.19z"
            fill="#FBBC05"
          />
          <path
            d="M24 48c6.48 0 11.93-2.13 15.89-5.81l-7.73-6c-2.15 1.45-4.92 2.3-8.16 2.3-6.26 0-11.57-4.22-13.47-9.91l-7.98 6.19C6.51 42.62 14.62 48 24 48z"
            fill="#34A853"
          />
          <path
            d="M0 0h48v48H0z"
            fill="none"
          />
        </g>
      </svg>
      <div
        className="hero-buttonLabel"
        onClick={[Function]}
      >
        Login with Google
      </div>
    </button>
  </div>
  <style>
    
            .hero {
                width: 100%;
                color: #333;
            }
            .hero-logo {
                padding-top: 80px;
                margin-bottom: 20px;
            }
            .logo img {
                width: 80px
            }
            .hero-title {
                margin: 0;
                width: 100%;
                line-height: 1.15;
                font-size: 60px;
                font-weight: 300;
            }
            .hero-logo,
            .hero-title,
            .hero-description {
                text-align: center;
            }
            .hero-row {
                margin: 80px 0 40px;
                display: flex;
                flex-direction: row;
                justify-content: space-around;
                padding: 32px 0 32px;
                background-color: #f7f7f7;
                border-bottom: 1px solid #e8e8e8;
                border-top: 1px solid #e8e8e8;
            }
            .hero button {
                display: inline-block;
                background: white;
                color: #444;
                border-radius: 5px;
                border: 1px solid #d5d5d5;
                white-space: nowrap;
                cursor: pointer;
                transition: background-color 220ms ease-out;
                display: flex;
                height: 48px;
                padding-left: 1.5rem;
                text-transform: uppercase;
                letter-spacing: 0.01rem;
            }
            hero button:hover {
                background: linear-gradient(to bottom, #f8f8f8, #f1f1f1);
            }
            .hero-buttonLabel {
                padding: 0 1rem;
            }
            
  </style>
</div>
`;
